src/util/configurations.rs

```diff
@@ -0,0 +1,83 @@
+//! This is the config struct used to pass in configurations into the channel. This is kept by the channel manager
+//! channel_limits is only passed in as a reference and only used when accepting or opening a channel
+//! channel options is cloned into channel because those are all specified unique channel settings. It uses channel managers one as a base
+
+/// This is the main user config
+/// This struct should contain all user customizable options as this is passed to the channel to be accessed
+#[derive(Clone, Debug)]
+pub struct UserConfig{
+	/// optional user specified channel limits
+	/// These are only used on startup of channels, and are referenced to a single instance
+	pub channel_limits : ChannelHandshakeLimits,
+	/// channel options are unique to each channel
+	pub channel_options : ChannelConfig,
+}
+
+impl UserConfig {
+	///default constructor, calls ChannelOptions and ChannelLimits constructors
+    pub fn new() -> Self{
+        UserConfig {
+            channel_limits : ChannelHandshakeLimits::new(),
+			channel_options : ChannelConfig::new(),
+        }
+    }
+}
+
+/// This struct contains all the optional bolt 2 channel limits.
+/// If the user wants to check a value, the value needs to be filled in, as by default they are not checked
+#[derive(Copy, Clone, Debug)]
+pub struct ChannelHandshakeLimits{
+	/// minimum allowed funding_satoshis
+	pub min_funding_satoshis :u64,
+	/// maximum allowed htlc_minimum_msat
+	pub max_htlc_minimum_msat : u64,
+	/// min allowed max_htlc_value_in_flight_msat
+	pub min_max_htlc_value_in_flight_msat : u64,
+	/// max allowed channel_reserve_satashis
+	pub max_channel_reserve_satoshis : u64,
+	/// min allowed max_accepted_htlcs
+	pub min_max_accepted_htlcs : u16,
+	/// min allowed dust_limit_satashis
+	pub min_dust_limit_satoshis : u64,
```

Comment:
Hmm, shouldn't we also have a max_dust_limit_satoshis? min really just kinda needs to be the network-standard dust limit but we also don't want our counterparty to just burn everything to fees always.

Reply:
I will add that, currently I just added wat the spec said.

Reply:
Hmm, fair, though it might be nice to have a separate limit - you may want the remote side to always keep a reserve that is "material value" but may want to ensure that the dust limit is something smaller so that HTLCs generate outputs.

Reply:
added max as well.

Had to change the constructer of the channel thou to change the value of the dust_limit if its a test as the estimator constructer supplies a 26 for the tests.